src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -253,33 +253,40 @@ export class DashboardAppController {
       navActions[TopNavIds.VISUALIZE]();
     };
 
-    const updateIndexPatterns = (container?: DashboardContainer) => {
-      if (!container || isErrorEmbeddable(container)) {
-        return;
-      }
-
-      let panelIndexPatterns: IndexPattern[] = [];
-      Object.values(container.getChildIds()).forEach((id) => {
-        const embeddableInstance = container.getChild(id);
-        if (isErrorEmbeddable(embeddableInstance)) return;
-        const embeddableIndexPatterns = (embeddableInstance.getOutput() as any).indexPatterns;
-        if (!embeddableIndexPatterns) return;
-        panelIndexPatterns.push(...embeddableIndexPatterns);
-      });
-      panelIndexPatterns = uniqBy(panelIndexPatterns, 'id');
-
-      if (panelIndexPatterns && panelIndexPatterns.length > 0) {
-        $scope.$evalAsync(() => {
-          $scope.indexPatterns = panelIndexPatterns;
+    const updateIndexPatternsOperator = pipe(
+      filter((container: DashboardContainer) => !!container && !isErrorEmbeddable(container)),
+      map((container: DashboardContainer) => {
+        let panelIndexPatterns: IndexPattern[] = [];
+        Object.values(container!.getChildIds()).forEach((id) => {
+          const embeddableInstance = container!.getChild(id);
```

Comment:
Mind extracting this block into a `getEmbeddablePanel` function?

i.e. `Object.values(container!.getChildIds()).forEach((id) => getEmbeddablePanel)`